experimental/c-cpp/build_generator.py

```diff
@@ -623,7 +623,23 @@ def match_build_heuristics_on_folder(abspath_of_target: str):
       AutogenScannerSH(),
   ]
 
-  for scanner in all_checks:
+  checks_to_test = []
+
+  print("Filtering out build scripts")
+  build_heuristics_to_analyse = os.getenv('BUILD_HEURISTICS', 'all')
+  if build_heuristics_to_analyse != 'all':
+    all_build_heuristics = build_heuristics_to_analyse.split(',')
+    for name in all_build_heuristics:
+      for check in all_checks:
+        if check.name == name:
+          checks_to_test.append(check)
+        else:
+          checks_to_test.append(check)
```

Comment:
Same statement under both `if` and `else`?

Reply:
Thanks!

Reply:
fixed